ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManager.java

```diff
@@ -1508,23 +1521,25 @@ private void ensurePartitionAbsenceOnNodeAndValidateCapacity(AmbryPartition part
      * @param partiallySealedReplicas {@link Collection} of partition names whose replicas are partially sealed.
      * @return ReplicaSealStatus object.
      */
-    private ReplicaSealStatus resolveReplicaSealStatus(String partitionName, Collection<String> sealedReplicas,
+    ReplicaSealStatus resolveReplicaSealStatus(String partitionName, Collection<String> sealedReplicas,
         Collection<String> partiallySealedReplicas) {
+      ReplicaSealStatus replicaSealStatus = ReplicaSealStatus.NOT_SEALED;
       if (clusterMapConfig.clusterMapEnablePartitionOverride && partitionOverrideInfoMap.containsKey(partitionName)) {
-        if (partitionOverrideInfoMap.get(partitionName)
-            .get(ClusterMapUtils.PARTITION_STATE)
-            .equals(ClusterMapUtils.READ_ONLY_STR)) {
-          return ReplicaSealStatus.SEALED;
+        replicaSealStatus =
+            ClusterMapUtils.partitionStateStrToReplicaSealStatus(partitionOverrideInfoMap.get(partitionName)
+                .get(ClusterMapUtils.PARTITION_STATE));
+        if (replicaSealStatus == ReplicaSealStatus.SEALED) {
+          return  replicaSealStatus;
         }
       } else {
         if (sealedReplicas.contains(partitionName)) {
           return ReplicaSealStatus.SEALED;
         }
       }
       if (partiallySealedReplicas.contains(partitionName)) {
```

Comment:
When do we use clusterMapEnablePartitionOverride to overwrite the partition to sealed?
Is that the manual operation used by SRE team?

Just wondering if we need clusterMapEnablePartitionOverride for partial sealed partition as well. If it's just used for SRE team to manual sealing, I guess we don't need the override for partial sealed as your current code.

Reply:
The cluster map partition is done manually. We add a list of partitions with their seal status in a znode of helix property store. This can be used to force set the status of a partition.

We don't need to override partial-sealed using cluster map override as of now. But in case we need it in the future, it's already supported. The map entry contains not just partition id, but also the state it's being overridden to. So for forcing partial-seal, we just need to mark the partition as "PRW" in the property store.